os/include/tinyara/sched.h

```diff
@@ -626,15 +632,20 @@ struct tcb_s {
 
 	struct xcptcontext xcp;		/* Interrupt register save area        */
 
-#ifdef CONFIG_APP_BINARY_SEPARATION
+#if defined(CONFIG_APP_BINARY_SEPARATION)
 	uint32_t ram_start;		/* Start address of RAM partition for this app */
 	uint32_t ram_size;		/* Size of RAM partition for this app */
 	uint32_t uspace;		/* User space object for app binary */
 	uint32_t uheap;			/* User heap object pointer */
-
 #ifdef CONFIG_ARM_MPU
-	uint32_t mpu_regs[3 * MPU_NUM_REGIONS];	/* We need 3 register values to configure each MPU region */
+#ifdef CONFIG_MPU_STACKGUARD
+	uint32_t mpu_regs[3 * (MPU_NUM_REGIONS + 1)]; /* We need 3 register values to configure each MPU region */
```

Comment:
This has the same comment as below, but it has different the element number of array. Please write the right comment.

Reply:
changed